packages/server/src/fhir/repo.ts

```diff
@@ -654,14 +649,33 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
       )?.profile;
       resource.meta = { ...resource.meta, profile: defaultProfiles };
     }
-
     if (!this.supportsInteraction(interaction, resourceType)) {
       throw new OperationOutcomeError(forbidden);
     }
+    return resource;
+  }
+
+  private async updateResourceImpl<T extends Resource>(
+    resource: T,
+    create: boolean,
+    versionId?: string
+  ): Promise<WithId<T>> {
+    const interaction = create ? AccessPolicyInteraction.CREATE : AccessPolicyInteraction.UPDATE;
+    const { resourceType, id } = resource;
+    resource = this.checkResourcePermissions(resource, interaction) as T;
```

Comment:
Hm... we try to avoid casts like this, especially deep in the bowels of the update code.

I suspect the reason you needed `as T` here is because `checkResourcePermissions` returns `WithId<T>` but `resource` is `T`.

I wonder if the better fix would be to change `resource` from `T` to `WithId<T>`